thoth/solver/python/python_solver.py

```diff
@@ -35,12 +35,20 @@
 
 
 class PythonReleasesFetcher(ReleasesFetcher):
+    """A releases fetcher based on PEP compatible simple API (also supporting Warehouse API)."""
     def __init__(self, source: Source):
         self.source = source
 
     def fetch_releases(self, package_name):
         package_name = self.source.normalize_package_name(package_name)
-        return package_name, self.source.get_package_versions(package_name)
+        releases = self.source.get_package_versions(package_name)
+        releases_with_index_url = [(item, self.index_url) for item in releases]
+        return package_name, releases_with_index_url
+
+    @property
+    def index_url(self):
+        """An URL to package source index from where releases are fetched."""
+        return self.source.url
```

Comment:
what if I do `PythonReleasesFetcher(None)` to init an instance?

Reply:
The signature of the constructor describes that there should be passed a source of type Source, not None:

```python
 __init__(self, source: Source):
```